src/command/FileCommands.cxx

```diff
@@ -233,3 +239,103 @@ handle_read_comments(Client &client, Request args, Response &r)
 
 	gcc_unreachable();
 }
+
+InputStreamPtr
+find_stream_art(const char *directory, Mutex &mutex, Cond &cond)
+{
+    static constexpr char const * art_names[] = {
+        "cover.png",
+        "cover.jpg",
+        "cover.tiff",
+        "cover.bmp"
+    };
+    
+    for(const auto name: art_names) {
+        const char *art_file = PathTraitsUTF8::Build(directory, name).c_str();
+        
+        try {
+            auto is = InputStream::OpenReady(art_file, mutex, cond);
+            return is;
+        } catch (const std::exception &e) {}
+    }
+    return nullptr;
+}
+
+static CommandResult
+read_stream_art(Response &r, const char *uri, size_t offset)
+{
+    const char *art_directory = PathTraitsUTF8::GetParent(uri).c_str();
```

Comment:
This will crash because the returned `std::string` will be freed after the statement, and the `c_str()` result points to that freed memory.